### DISCUSSION TITLE
How to access `app_id` and `partition_key` inside an action?

### DISCUSSION THREAD

**Discussion by shun-liang** [DAGWorks-Inc/burr#445] [Help + Q&A] [answered]
Dear burr developers

Sorry if it's obivous, but in the documentation I haven't found any information about how to access `app_id` and `partition_key` inside an action. Can you give me some information on this? Many thanks.

**Comment by skrawcz** [comment, marked as answer]
Yes, you're right this isn't in the docs. We need to add that.  You can access it in the action by declaring a `__context: ApplicationContext` parameter, and then accessing it like so:

```python
from burr.core import action, State, ApplicationContext

@action(reads=["..."], writes=["..."])
def my_action(state: State, __context: ApplicationContext) -> State:
    print(__context.app_id, __context.partition_key, __context.sequence_id)
    ...
```
See [this notebook for example usage](https://github.com/DAGWorks-Inc/burr/blob/afc0fc031a7e775bb82d0110b367e25666bd0f97/examples/parallelism/notebook.ipynb#L502).

**Reply by skrawcz** [reply]
Added note in the [docs here for now](https://burr.dagworks.io/concepts/state-persistence/#state-keys).

**Comment by shun-liang** [comment]
Thanks for the answer @skrawcz . Can you please also add an example achieving the same but with a class-based action?

**Reply by skrawcz** [reply]
ah yep - found a minor bug with python name mangling. Will fix today and push it out later tonight.

**Reply by skrawcz** [reply]
I added an example to the bottom of https://burr.dagworks.io/concepts/actions/#class-based-actions 
Note this requires the latest `burr==0.34.0`.
